regiontag_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package samples
+
+import (
+	"bufio"
+	"bytes"
+	"log"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+func TestRegionTags(t *testing.T) {
+	type RegionTag struct {
+		tag  string
+		path string
+	}
+	errs := []RegionTag{}
+
+	cmd := exec.Command("go", "list", "./...")
+	out, err := cmd.Output()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(string(out)))
+	for scanner.Scan() {
+		path := scanner.Text()
+
+		goDoc := exec.Command("go", "doc", path)
+		grep := exec.Command("grep", "START")
```

Comment:
I'm concerned about Windows users (or anyone without `grep` installed). Maybe this could "grep" in pure Go? Could maybe do something similar to above (untested):
```go
goDoc := exec.Command("go", "doc", path)
docOut, err := goDoc.StdoutPipe()
if err != nil {
  t.Fatalf(err)
}
docScanner := bufio.NewScanner(docOut)
for docScanner.Scan() {
  text := scanner.Text()
  if !strings.Contains(text, "[START") {
    continue
  }
  ...
}
```

Reply:
I'd lean towards using `go/doc`. @danicat, did you know about this package?

https://golang.org/pkg/go/doc/

Reply:
I keep forgetting that people still use windows... lol... will fix that :)

Reply:
I was not familiar with that package @broady. I'll have a look at it.

Reply:
In my newest commit I have removed the grep dependency as suggested, but I haven't used go/doc yet. `doc.New` requires a package in ast format and I'm not sure of the right steps to load it in that form.